CODE_STYLE.md

```diff
@@ -0,0 +1,3293 @@
+   def some_method
+     # body omitted
+   end
+
+   # bad
+   def some_method_with_parameters param1, param2
+     # body omitted
+   end
+
+   # good
+   def some_method_with_parameters(param1, param2)
+     # body omitted
+   end
+   ```
+
+* <a name="parallel-assignment"></a>
+    Avoid the use of parallel assignment for defining variables. Parallel
+    assignment is allowed when it is the return of a method call, used with
+    the splat operator, or when used to swap variable assignment. Parallel
+    assignment is less readable than separate assignment. It is also slightly
+    slower than separate assignment.
+<sup>[[link](#parallel-assignment)]</sup>
+
+  ```Ruby
+  # bad
+  a, b, c, d = 'foo', 'bar', 'baz', 'foobar'
+
+  # good
+  a = 'foo'
+  b = 'bar'
+  c = 'baz'
+  d = 'foobar'
+
+  # good - swapping variable assignment
+  # Swapping variable assignment is a special case because it will allow you to
+  # swap the values that are assigned to each variable.
+  a = 'foo'
+  b = 'bar'
+
+  a, b = b, a
+  puts a # => 'bar'
+  puts b # => 'foo'
+
+  # good - method return
+  def multi_return
+    [1, 2]
+  end
+
+  first, second = multi_return
+
+  # good - use with splat
+  first, *list = [1,2,3,4]
+
+  hello_array = *"Hello"
+
+  a = *(1..3)
+  ```
+
+* <a name="no-for-loops"></a>
+    Do not use `for`, unless you know exactly why. Most of the time iterators
+    should be used instead. `for` is implemented in terms of `each` (so
+    you're adding a level of indirection), but with a twist - `for`
+    doesn't introduce a new scope (unlike `each`) and variables defined
+    in its block will be visible outside it.
+<sup>[[link](#no-for-loops)]</sup>
+
+  ```Ruby
+  arr = [1, 2, 3]
+
+  # bad
+  for elem in arr do
+    puts elem
+  end
+
+  # note that elem is accessible outside of the for loop
+  elem # => 3
+
+  # good
+  arr.each { |elem| puts elem }
+
+  # elem is not accessible outside each's block
+  elem # => NameError: undefined local variable or method `elem'
+  ```
+
+* <a name="no-then"></a>
+  Do not use `then` for multi-line `if/unless`.
+<sup>[[link](#no-then)]</sup>
+
+  ```Ruby
+  # bad
+  if some_condition then
+    # body omitted
+  end
+
+  # good
+  if some_condition
+    # body omitted
+  end
+  ```
+
+* <a name="same-line-condition"></a>
+  Always put the condition on the same line as the `if`/`unless` in a
+  multi-line conditional.
+<sup>[[link](#same-line-condition)]</sup>
+
+  ```Ruby
+  # bad
+  if
+    some_condition
+    do_something
+    do_something_else
+  end
+
+  # good
+  if some_condition
+    do_something
+    do_something_else
+  end
+  ```
+
+* <a name="ternary-operator"></a>
+  Favor the ternary operator(`?:`) over `if/then/else/end` constructs.
+  It's more common and obviously more concise.
+<sup>[[link](#ternary-operator)]</sup>
+
+  ```Ruby
+  # bad
+  result = if some_condition then something else something_else end
+
+  # good
+  result = some_condition ? something : something_else
```

Comment:
I'm on the fence. I tend to avoid ternary in practice, especially in ruby, mostly due to the presence of predicate-style method names (ones ending in `?`). This looks weird to me:

```ruby
foo = something? ? bar : baz
```

And for symbolic values, this also looks weird to me:

```ruby
foo = something ? :bar : :baz
```

I am in favor of not using ternary operator if at all possible.